gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/ddl/catalog/GfxdSystemProcedures.java

```diff
@@ -2425,14 +2425,17 @@ public static void COMMIT_SNAPSHOT_TXID(String txId) throws SQLException, Standa
     int uniqId = Integer.parseInt(st.nextToken());
     TXId txId1 = TXId.valueOf(memberId, uniqId);
     TXStateInterface txState = Misc.getGemFireCache().getCacheTransactionManager().getHostedTXState(txId1);
-    Misc.getGemFireCache().getCacheTransactionManager().masqueradeAs(txState);
-    Misc.getGemFireCache().getCacheTransactionManager().commit();
+    LanguageConnectionContext lcc = ConnectionUtil.getCurrentLCC();
+    GemFireTransaction tc = (GemFireTransaction)lcc.getTransactionExecute();
+    tc.clearActiveTXState(false, true);
+    Misc.getGemFireCacheNoThrow().getCacheTransactionManager().masqueradeAs(txState);
+    Misc.getGemFireCacheNoThrow().getCacheTransactionManager().commit();
```

Comment:
If cache is just closing and this returns null? Better to get from tc.getTransactionManager()